### PR TITLE
Explosions now cause camera shake based on distance

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl:
add: Explosions now cause camera shake based on intensity and distance
/:cl:

### CODE REVIEW

**Comment by Buggy123** [comment]
We Michael Bay now.

**Comment by MrDoomBringer** [comment]
eventually came faster than i thought :o

File: code/datums/explosion.dm [hunk 1]
```diff
@@ -119,11 +121,13 @@ GLOBAL_LIST_EMPTY(explosions)
 				// If inside the blast radius + world.view - 2
 				if(dist <= round(max_range + world.view - 2, 1))
 					M.playsound_local(epicenter, null, 100, 1, frequency, falloff = 5, S = explosion_sound)
+					shake_camera(M, 50, orig_max_distance - dist)
```
Comment (MrStonedOne) [inline code/datums/explosion.dm]: 5 seconds of shake MIGHT be a bitch much.

Also time of shake and amount of shake *might* need to scale together.

Comment (MrStonedOne) [reply code/datums/explosion.dm]: also this might need a cap on it, as atm it doesn't scale so well when you take into account admin explosions.

Comment (Cyberboss) [reply code/datums/explosion.dm]: Yeah i need to fuck around and find some good numbers

**Comment by deathride58** [comment]
Players might complain about how the extra screen shake makes the game literally unplayable

You should probably port Citadel's screenshake preference so you guys don't get ten thousand angry snowflakes spamming OOC when this gets merged.

**Comment by MrMaclang** [comment]
time to cause explotions on cammeras

**Comment by ninjanomnom** [comment]
Stop leaving your stinky WIP everywhere @Cyberboss

**Comment by Cyberboss** [comment]
I forgot this existed

**Comment by ShizCalev** [comment]
Still wip?

**Comment by MrStonedOne** [comment]
@Cyberboss rebase to fix travis

**Review by ninjanomnom — APPROVED** [review]

**Comment by vuonojenmustaturska** [comment]
You might've actually wanted to test this before merging it

**Pull request merged**